tests/drivers/mspi/error_cases/src/main.c

```diff
@@ -0,0 +1,686 @@
+};
+
+static const struct mspi_cfg default_hw_cfg = {
+	.channel_num      = 0,
+	.op_mode          = DT_ENUM_IDX_OR(MSPI_BUS_NODE, op_mode, MSPI_OP_MODE_CONTROLLER),
+	.duplex           = MSPI_FULL_DUPLEX,
+	.dqs_support      = DT_PROP_OR(MSPI_BUS_NODE, dqs_support, false),
+	.sw_multi_periph  = DT_PROP_OR(MSPI_BUS_NODE, sw_multi_periph, false),
+	.ce_group         = ce_gpios,
+	.num_ce_gpios     = ARRAY_SIZE(ce_gpios),
+	.num_periph       = DT_CHILD_NUM(MSPI_BUS_NODE),
+	.max_freq         = DT_PROP(MSPI_BUS_NODE, clock_frequency),
+	.re_init          = true,
+};
+
+uint8_t memc_write_buffer[256];
+
+static struct mspi_xfer_packet packet1[] = {
+	{
+		.dir          = MSPI_TX,
+		.cmd          = 0x12,
+		.address      = 0,
+		.num_bytes    = 256,
+		.data_buf     = memc_write_buffer,
+		.cb_mask      = MSPI_BUS_NO_CB,
+	},
+};
+
+static struct mspi_xfer default_xfer = {
+	.async            = false,
+	.xfer_mode        = MSPI_DMA,
+	.tx_dummy         = 0,
+	.cmd_length       = 1,
+	.addr_length      = 3,
+	.priority         = 1,
+	.packets          = (struct mspi_xfer_packet *)&packet1,
+	.num_packet       = sizeof(packet1) / sizeof(struct mspi_xfer_packet),
+};
+
+static int ret;
+
+/* Helper function. */
+void check_mspi_config(struct mspi_cfg *hw_cfg, int expected)
+{
+
+	const struct mspi_dt_spec spec = {
+		.bus    = mspi_bus,
+		.config = *hw_cfg,
+	};
+
+	ret = mspi_config(&spec);
+	zassert_equal(ret, expected, "Got unexpected %d instead of %d", ret, expected);
+}
+
+ZTEST(mspi_error_cases, test_01a_MSPI_OP_MODE_CONTROLLER)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.op_mode = MSPI_OP_MODE_CONTROLLER;
+
+	check_mspi_config(&hw_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_01b_MSPI_OP_MODE_PERIPHERAL)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.op_mode = MSPI_OP_MODE_PERIPHERAL;
+
+	check_mspi_config(&hw_cfg, -ENOTSUP);
+}
+
+ZTEST(mspi_error_cases, test_02a_MSPI_HALF_DUPLEX)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.duplex = MSPI_HALF_DUPLEX;
+
+	check_mspi_config(&hw_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_02b_MSPI_FULL_DUPLEX)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.duplex = MSPI_FULL_DUPLEX;
+
+	check_mspi_config(&hw_cfg, -ENOTSUP);
+}
+
+ZTEST(mspi_error_cases, test_03a_dqs_support_true)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.dqs_support = true;
+
+	check_mspi_config(&hw_cfg, -ENOTSUP);
+}
+
+ZTEST(mspi_error_cases, test_03b_dqs_support_false)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.dqs_support = false;
+
+	check_mspi_config(&hw_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_04a_sw_multi_periph_true)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.sw_multi_periph = true;
+
+	check_mspi_config(&hw_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_05a_num_ce_gpios_max)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.num_ce_gpios = UINT32_MAX;
+
+	check_mspi_config(&hw_cfg, 0);
```

Comment:
```suggestion
	check_mspi_config(&hw_cfg, ENOTSUP);
```
Max 5 CE pins are supported.

Reply:
I set -EINVAL to be consistent with above comment.